src/Sylius/Bundle/ApiBundle/CommandHandler/Checkout/ChoosePaymentMethodHandler.php

```diff
@@ -78,6 +79,6 @@ public function __invoke(ChoosePaymentMethod $choosePaymentMethod): OrderInterfa
             return $cart;
         }
 
-        throw new \InvalidArgumentException('Payment method can not be set');
+        throw new PaymentMethodCannotBeSetException('Payment method can not be set');
```

Comment:
If you decide to provide suggestions from `src/Sylius/Bundle/ApiBundle/Changer/PaymentMethodChanger.php`, they apply also here.